src/main/java/com/google/cloud/tools/gradle/appengine/core/DeployExtension.java

```diff
@@ -31,9 +31,9 @@
 
   // named gradleProject to disambiguate with deploy parameter "project"
   private final Project gradleProject;
+  private final List<File> deployables;
```

Comment:
Could this be an immutable list? http://google.github.io/guava/releases/23.0/api/docs/com/google/common/collect/ImmutableList.html

Reply:
I don't think so, the list is cleared/repopulated in deployAll.

Reply:
yeah, maybe deployAll could just give the copy constructor a fully populated list.